src/exif/service.rs

```diff
@@ -0,0 +1,172 @@
+                "Failed to parse {message_data} as a u32! Exception is {e:?}\n{}",
+                Backtrace::force_capture()
+            );
+            // Invalid message, don't re-queue
+            return true;
+        }
+    };
+
+    // Get file metadata from database
+    let con = open_connection();
+    let file_record = match file_repository::get_file(id, &con) {
+        Ok(f) => f,
+        Err(e) => {
+            log::debug!(
+                "Failed to get file from database when processing EXIF data. File id [{id}]; error is {e:?}",
+            );
+            con.close().unwrap();
+            // File doesn't exist in DB, don't requeue
+            return true;
+        }
+    };
+
+    // Only process image and video files
+    match file_record.file_type {
+        FileTypes::Image | FileTypes::Video => {}
+        _ => {
+            con.close().unwrap();
+            // Not an image/video, no EXIF to process
+            return true;
+        }
+    }
+
+    // Get file path on disk
+    let path = match get_file_path(id) {
+        Ok(p) => {
+            let path = PathBuf::from(file_dir()).join(p);
+            path.to_string_lossy().into_owned()
+        }
+        Err(_) => {
+            log::debug!("File id {id} not found on disk, skipping EXIF processing");
+            con.close().unwrap();
+            // File is no longer on disk, don't requeue
+            return true;
+        }
+    };
+
+    // Parse EXIF date or use current date as fallback
+    let create_date =
+        parse_exif_date(&path).unwrap_or_else(|| chrono::offset::Local::now().naive_local());
+
+    // Update file record in database with the extracted date
+    let update_result = con.execute(
+        "UPDATE FileRecords SET dateCreated = ?1 WHERE id = ?2",
+        rusqlite::params![create_date, id],
+    );
+
+    con.close().unwrap();
+
+    match update_result {
+        Ok(_) => {
+            log::debug!("Successfully updated creation date for file id {id}");
+            true
+        }
+        Err(e) => {
+            log::error!(
+                "Failed to update creation date for file id {id}: {e:?}\n{}",
+                Backtrace::force_capture()
+            );
+            // Database error, might be temporary, re-queue
+            false
+        }
+    }
+}
+
+/// Queues all image and video files in the database for EXIF processing.
+/// This function should be called on startup if the exif processing flag is not set.
+pub fn mass_exif_process() {
+    use crate::queue::publish_message;
+
+    let con = open_connection();
+    let all_files = match file_repository::get_all_files(&con) {
+        Ok(files) => files,
+        Err(e) => {
+            log::error!(
+                "Failed to retrieve files for mass EXIF processing: {e:?}\n{}",
+                Backtrace::force_capture()
+            );
+            con.close().unwrap();
+            return;
+        }
+    };
+    con.close().unwrap();
```

Comment:
you don't need 2 calls to close the connection. Close it before the match statement but after the initial repository call to get files
```rs
let con = open_connection();
let all_files = file_repository::get_all_files(&con);
con.close().unwrap();
let all_files = match all_files // ...
```